Add support for Azure availability sets (refactored) (#1582)
epiphany-platform/epiphany: docs/home/howto/CLUSTER.md

## RabbitMQ installation and setting

To install RabbitMQ in single mode just add rabbitmq role to your data.yaml for your server and in general roles section. All configuration on RabbitMQ - e.g. user other than guest creation should be performed manually.

## Howto use azure scalability set feature

In your cluster yaml config declare several (if required) object of kind `infrastructure/availability-set` like
example below, change the `name` field as you wish.

```yaml
---
kind: infrastructure/availability-set
name: kube-node-availability-set
specification:
# The "name" attribute is generated automatically according to Epiphany's naming conventions
  availability_set: kube-node  # Short and simple name is preferred
  platform_fault_domain_count: 2
  platform_update_domain_count: 5
  managed: 'true'
provider: azure
```

> Then in the corresponding `components` section of the kind `kind: epiphany-cluster` set it as well.

Reviewer: ```suggestion
Then set it also in the corresponding `components` section of the `kind: epiphany-cluster` doc.
```

```yaml
  components:
    kafka:
      count: 0
    kubernetes_master:
      count: 1
    kubernetes_node:
# This line tells we generate the availability-set terraform template
      availability_set: kube-node  # Short and simple name is preferred
      count: 2
```

> The full minimum cluster is listed below. Note that you can set different type of machine within different
> availability set, eg. all kubenetes master in one availability set, all kubernetes nodes in another one, also
> kafka etc.. .

Reviewer: ```suggestion
The full minimum cluster is listed below. Note that you can set different type of machine within different
availability set, e.g. all `kubernetes_master` in one availability set, all `kubernetes_node` in another one, also
`kafka` etc.
```

## RabbitMQ installation and setting

To install RabbitMQ in single mode just add rabbitmq role to your data.yaml for your server and in general roles section. All configuration on RabbitMQ - e.g. user other than guest creation should be performed manually.

## How to use Azure availability sets

In your cluster yaml config declare as many as required objects of kind `infrastructure/availability-set` like
in the example below, change the `name` field as you wish.

```yaml
---
kind: infrastructure/availability-set
name: kube-node  # Short and simple name is preferred
specification:
# The "name" attribute is generated automatically according to Epiphany's naming conventions
  platform_fault_domain_count: 2
  platform_update_domain_count: 5
  managed: true
provider: azure
```

Then set it also in the corresponding `components` section of the `kind: epiphany-cluster` doc.

```yaml
  components:
    kafka:
      count: 0
    kubernetes_master:
      count: 1
    kubernetes_node:
# This line tells we generate the availability-set terraform template
      availability_set: kube-node  # Short and simple name is preferred
      count: 2
```

> The full minimum cluster is listed below. Note that you can set different type of machine within different
> availability set, e.g. all `kubernetes_master` in one availability set, all `kubernetes_node` in another one, also
> `kafka` etc.

Reviewer: ```suggestion
The example below shows a complete configuration. Note that it's recommended to have a dedicated availability set for each clustered component.
```
